core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_test.go

```diff
@@ -0,0 +1,1510 @@
+        }
+        cc.assetLockMap[key] = val
+        return shim.Success(nil)
+    }
+    if function == "LockFungibleAsset" {    // We are only going to lock once or twice in each unit test function, so bookkeeping doesn't need to be thorough
+        assetAgreement := &common.FungibleAssetAgreement{}
+        _ = proto.Unmarshal([]byte(args[0]), assetAgreement)
+        key := assetAgreement.Type + ":" + strconv.Itoa(int(assetAgreement.NumUnits))
+        val := string(caller) + ":" + assetAgreement.Recipient
+        if cc.fungibleAssetUnlockedCount[assetAgreement.Type] < int(assetAgreement.NumUnits) {
+            return shim.Error(fmt.Sprintf("Requested %d units of asset type %s; only %d available", assetAgreement.NumUnits, assetAgreement.Type, cc.fungibleAssetUnlockedCount[assetAgreement.Type]))
+        }
+        cc.fungibleAssetLockMap[key] = val
+        cc.fungibleAssetUnlockedCount[assetAgreement.Type] = cc.fungibleAssetUnlockedCount[assetAgreement.Type] - int(assetAgreement.NumUnits)
+        return shim.Success(nil)
+    }
+    if function == "IsAssetLocked" {
+        expectedKey := args[0] + ":" + args[1]
+        expectedVal := args[3] + ":" + args[2]
+        if cc.assetLockMap[expectedKey] == expectedVal {
+            return shim.Success([]byte("true"))
+        } else {
+            return shim.Success([]byte("false"))
+        }
+    }
+    if function == "IsFungibleAssetLocked" {
+        expectedKey := args[0] + ":" + args[1]
+        expectedVal := args[3] + ":" + args[2]
+        if cc.fungibleAssetLockMap[expectedKey] == expectedVal {
+            return shim.Success([]byte("true"))
+        } else {
+            return shim.Success([]byte("false"))
+        }
+    }
+    if function == "UnlockAsset" {
+        expectedKey := args[0] + ":" + args[1]
+        expectedVal := string(caller) + ":" + args[2]
+        if cc.assetLockMap[expectedKey] == "" {
+            return shim.Error(fmt.Sprintf("No asset of type %s and ID %s is locked", args[0], args[1]))
+        } else if cc.assetLockMap[expectedKey] != expectedVal {
+            return shim.Error(fmt.Sprintf("Cannot unlock asset of type %s and ID %s as it is locked by %s for %s", args[0], args[1], string(caller), args[2]))
+        } else {
+            delete(cc.assetLockMap, expectedKey)
+            return shim.Success(nil)
+        }
+    }
+    if function == "UnlockFungibleAsset" {
+        expectedKey := args[0] + ":" + args[1]
+        expectedVal := string(caller) + ":" + args[2]
+        if cc.fungibleAssetLockMap[expectedKey] == "" {
+            return shim.Error(fmt.Sprintf("No asset of type %s and Units %s is locked", args[0], args[1]))
+        } else if cc.fungibleAssetLockMap[expectedKey] != expectedVal {
+            return shim.Error(fmt.Sprintf("Cannot unlock asset of type %s and Units %s as it is locked by %s for %s", args[0], args[1], string(caller), args[2]))
+        } else {
+            delete(cc.fungibleAssetLockMap, expectedKey)
```

Comment:
We have to reset the `fungibleAssetUnlockedCount` ideally here (e.g., `cc.fungibleAssetUnlockedCount[args[0]] = cc.fungibleAssetUnlockedCount[args[1]] + int(args[2])`).

Reply:
Same answer as above.